hw_numpy/numpy_challenge.py

```diff
@@ -0,0 +1,43 @@
+if __name__ == "__main__":
+  import numpy as np
+  a = np.ones((3, 5))
+  b = np.arange(0, 60, 5).reshape(3, 4)
+  c = np.random.random((10, 10))
+
+# matrix multiplication
+def matrix_multiplication(first_arr, second_arr):
+   if first_arr.shape[1] == second_arr.shape[0]:
+     return(np.dot(first_arr, second_arr))
+
+# check if it is possible to multiply arrays
+def multiplication_check(arr):
+  b = np.vectorize(lambda x: x.shape)(np.array(list(map(lambda x: np.array(x), arr))))
+  if all(b[0][1:] == b[1][:-1]) == True:
+    return True
+  else:
+    return False
+
+# matrix multiplication from list
+def multiply_matrices(arr):
+  if multiplication_check(arr) == True:
+    for i in range(len(arr)-1):
+      a = np.dot(arr[i], arr[i+1])
+    return(a)
+  else:
+    return None
```

Comment:
1. Здесь ты по сути просто перемножаешь по две соседних матрицы в списке, но результат у тебя нигде не накапливается, поэтому ответ получается неправильный
2. В питоне принято делать отступы кратные 4 пробелам.
3. Всю работу может сделать функция `np.multidot`
4. Нейминг переменных
```suggestion
def multiply_matrices(matrices_list):
  if multiplication_check(matrices_list):
    result_matrix = matrices_list[0]
    for i in range(1, len(result_matrix)):
      result_matrix = np.dot(result_matrix, arr[i])
    return a 
  else:
    return None
```